ADR/ADR-Guidelines.md

```diff
@@ -0,0 +1,140 @@
+# ECMWF Architectural Decision Records (ADR) Guidelines
+
+This document provides a Guidelines and a template for creating Architectural Decision Records (ADRs). Each ADR should be a standalone document that captures a specific architectural decision, its context, and its consequences.
+
+## Template for Architectural Decision Records (ADRs)
+
+Please find the template for creatinsg ADRs in the [ADR-Template.md](./ADR-Template.md) file. This template should be used for all ADRs to ensure consistency and clarity.
+
+## Purpose
+
+Architectural Decision Records (ADRs) are lightweight documents that capture important architectural decisions made during the development of ECMWF systems. They provide historical context, rationale, and consequences of decisions to help current and future team members understand the evolution of our architecture.
+
+ADRs are living documents that serve the team. We should focus on creating useful, accessible records that will help current and future team members understand and build upon the architectural decisions that shape ECMWF's systems.
+
+## When to Write an ADR
+
+Create an ADR when making decisions that:
+
+- **Significantly impact system architecture** - Architectural choices, framework selections, major design patterns
+- **Affect multiple teams or systems** - Cross-cutting concerns, shared libraries, integration patterns
+- **Have long-term consequences** - Technology stack decisions, data models, security approaches, scalability strategies
+- **Require justification and analysis** - Trade-offs between options, risk assessments, cost-benefit
+- **Involve trade-offs between alternatives** - For example performance vs. maintainability, cost vs. functionality
+- **Establish important conventions** - Coding standards, deployment processes, monitoring strategies
+- **Address compliance or regulatory requirements** - Adherance to standards like OGC and Data governance, but also security policies and identity managment
+
+## Writing Guidelines
+
+### General Principles
+
+- **Be concise but comprehensive** - Provide enough detail for understanding without unnecessary verbosity
+- **Focus on the decision, not implementation details** - Explain what and why, not how
+- **Use clear, jargon-free language** - Ensure accessibility to team members with different backgrounds
+- **Be objective** - Present facts and analysis rather than opinions
+- **Include dissenting views** - Acknowledge alternative perspectives when relevant
+
+### Section-Specific Guidelines
```

Comment:
there seems some overlap between this part and the template. For example, the `Status` enum does not need repetition, in the sense that the keywords in the template are self-explanatory

similarly, Context could be moved/merged into the template.

I'm thinking about the user flow -- we should read Guidelines ~once, and then only go over template when we fill out an ADR proposal. Thus I'd keep Guideline very high level, general, without this level of detailed instructions

Reply:
Yes there is definitely an overlap. But where to put the explanations? where to check you have completeness? 
I think having the a full description of what is expected belongs in the guidelines and it helps provide also a good sense of what is expected from an ADR. The template is more there to copy paste a file and edit in place, then refer to the guidelines if one has questions for clarity.

Reply:
I think the template should be shortened to one sentence per heading, this document gives the actual guidelines, the template serves only as a quick reminder when drafting up a new ADR.

Reply:
@Ozaq you make a good point and @tmi idea of a checklist is also attractive. I'll try update the template to be short and sweet, more in the line of checklist.
We keep the guidelines with the longer explanation.